action.yaml

```diff
@@ -146,11 +150,47 @@ runs:
       env:
         from_tags: ${{ steps.cache-from.outputs.tags }}
         to_tags: ${{ steps.cache-to.outputs.tags }}
+    - name: Process cache mount IDs
+      id: cache-mount
+      shell: bash
+      run: |
+        # Process cache mount IDs
+        ids_json="$(jq -R 'select(. != "")' <<<"${cache_mount_ids}" | jq -sc .)"
+
+        # We fully support user provided IDs which are absolute paths.
+        bind_root="$(readlink -f "${bind_root:?}")"
+        cache_map_json="$(jq --arg bind_root "${bind_root:?}" 'map(. as $id | {"key": ($bind_root + "/" + $id), "value": {"id": $id, "target": ("/tmp/" + $id)}}) | from_entries' <<<"${ids_json}")"
+
+        # Specify our multiline output using GH action flavored heredocs
+        # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+        {
+            echo "cache-map<<EOF"
+            jq <<<"${cache_map_json}"
+            echo "EOF"
+        } | tee -a "$GITHUB_OUTPUT"
+      env:
+        cache_mount_ids: ${{ inputs.cache-mount-ids }}
+        bind_root: ${{ github.action_path }}/cache-mount
+    - name: Retrieve Docker cache mount bundle
+      uses: actions/cache@v4
+      with:
+        path: |
+          ${{ github.action_path }}/cache-mount
+        key: docker-build-cache-mount-${{ hashFiles(format('{0}/**', inputs.context)) }}
```

Comment:
I'm confused about this: we're hashing _everything_ in the build context, so won't invalidate the cache if _anything_ changes?  is that really what we want?

Reply:
See: https://github.com/beacon-biosignals/docker-build/pull/24#discussion_r2038370941